docs/learn-transaction-fees.md

```diff
@@ -0,0 +1,81 @@
+
+Block producers prioritize transactions based on each transaction's total fee. Since a portion of the fee will go to the block producer, producers will include the transactions with the highest fees to maximize their reward.
+
+## Fee Adjustment
+
+Transaction volume on blockchains is highly irregular, and therefore transaction fees need a mechanism to adjust. However, users should be able to predict transaction fees.
+
+Polkadot uses a slow-adjusting fee mechanism with tips to balance these two considerations. In addition to block _limits,_ Polkadot also has a block fullness _target._ Fees increase or decrease for the next block based on the fullness of the current block relative to the target. The per-weight fee can change up to 30% in a 24 hour period. This rate captures long-term trends in demand, but not short-term spikes. To consider short term spikes, Polkadot uses tips on top of the base, byte, and weight fees. Users can optionally add a tip to the fee to give the transaction a higher priority.
+
+## Shard Transactions
+
+The transactions that take place within Polkadot's shards - parachains and parathreads - do not incur Relay Chain transaction fees. Users of shard applications do not even need to hold DOT tokens, as each shard has its own economic model and may or may not have a token. There are, however, situations where shards themselves make transactions on the Relay Chain.
+
+[Parachains](learn-parachains) have a dedicated slot on the Relay Chain for execution, so their collators do not need to own DOTs in order to include blocks. The parachain will make some transactions itself, for example, opening or closing an [XCMP](learn-crosschain) channel, participating in an [auction](learn-auction) to renew its slot, or upgrading its runtime. Parachains have their own accounts on the Relay Chain and will need to use those funds to issue transactions on the parachain's behalf.
+
+[Parathreads](learn-parathreads) will also make all the same transactions that a parachain might. In addition, the collators need to participate in an auction every block to progress their chain. The collators will need to have DOTs to participate in these auctions.
+
+## Other Resource Limitation Strategies
+
+Transaction weight must be computable prior to execution, and therefore can only represent fixed logic. Some transactions warrant limiting resources with other strategies. For example:
+
+- Bonds: Some transactions, like voting, may require a bond that will be returned later, or slashed if the voter tries anything malicious.
+- Burns: A transaction may burn funds internally based on its logic. For example, a transaction may burn funds from the sender if it creates new storage entries, thus increasing the state size.
```

Comment:
I'd like more info about this - are all write tx burning part of the fee or just some? If some, which for example?

Reply:
Nope, and actually we just removed the only example that existed. `transfer` used to take an extra fee if an account was created. Gav took it out with some other method (and the existential deposit is a form of bonding). I guess this could come out of the Polkadot RC wiki since we're not doing it anymore. Still is an option for designing a chain.

Reply:
As an example, this is the Balances pallet from a few weeks ago: https://github.com/paritytech/substrate/blob/d2bb42d314bbdcc48c0cf01cb59f7ab10294ebad/frame/balances/src/lib.rs#L989